### PR TITLE
Add validator for enrolment classes

### PR DESCRIPTION
**Pull request by danielchen115**
### Notion ticket link

[Validate that enrolled_class and preferred_class are in the enrolment's session](https://www.notion.so/uwblueprintexecs/Validate-that-enrolled_class-and-preferred_class-are-in-the-enrolment-s-session-e7ec5c044eef45f6a6d1d8d2580a2691)

### Implementation description

- Added validator for enrolments
- validator checks if its preferred_class and enrolled_class are related to its session

### Steps to test

1. Try to create an enrolment where its preferred/enrolled class is not in its session

### What should reviewers focus on?

- should fail with an error

### CODE REVIEW

File: enrolments/models.py [hunk 1]
```diff
@@ -67,3 +67,7 @@ class Enrolment(models.Model):
         blank=True,
         on_delete=models.SET_NULL,
     )
+
+    def clean(self):
+        validate_enrolment(self)
+        return super().clean()
```
Comment (Tayef-Shah) [inline enrolments/models.py:74]: nit: can we get a new line at the end of file?

**Review by Tayef-Shah — APPROVED** [review]
lgtm

File: enrolments/tests/test_validators.py [hunk 2]
```diff
@@ -1,10 +1,75 @@
...
+        class_no_sessions = Class.objects.create(
+            name="Class with no sessions",
+            session=None,
+            facilitator=self.user,
+            attendance={},
+        )
+        self.assertIsNone(
+            validate_class_in_session(self.class_in_session, self.session)
+        )
+        self.assertRaises(
```
Comment (Tayef-Shah) [inline enrolments/tests/test_validators.py:44]: might want to also test a class with a different session in addition to a class without a session

Comment (danielchen115) [reply enrolments/tests/test_validators.py:44]: added :)

File: enrolments/models.py (hunk 1 above)
Comment (danielchen115) [reply enrolments/models.py:74]: hmm for some reason it looks fine in my ide but not on github... I think it should still be there?

**Pull request merged**